tests/virt_autotest/virt_utils.pm

```diff
@@ -563,4 +571,43 @@ sub check_guest_disk_type {
     }
 }
 
+#return 1 if it is a fv guest judging by name
+#feel free to extend to support more cases
+sub is_fv_guest {
+    my $guest = shift;
+    return if ($guest =~ /\bfv\b/ || $guest =~ /\bhvm\b/);
+}
+
+#return 1 if it is a pv guest judging by name
+#feel free to extend to support more cases
+sub is_pv_guest {
+    my $guest = shift;
+    return if ($guest =~ /\bpv\b/);
+}
+
+#check host to make sure it works well
+#welcome everybody to extend this function
+sub check_host {
+    return 0;
+}
+
+#check guest to make sure it works well
+#welcome everybody to extend this function
+sub check_guest {
+    my $vm = shift;
+
+    #check if guest is still alive
+    if (script_output("virsh domstate $vm", 30) ne "running") {
+        script_run "virsh list --all";
+        return 1;
```

Comment:
I do not think return 0 or 1 is best practice. I think you should use return true or false.

Reply:
do you all think which value can flag the guest is in good state of check_guest(), true/1 or false/0? I am afraid of confusion.

Reply:
But 1 or 0 is much easier to cause confusion, 1 does not always mean true either in all programming language.  I think you should avoid confusion by using true or false instead the meanings of which are constant. Or you can choose your own more meaningful words, for example, OK or NOK.

Reply:
change to return the actual state of the host or guest.